Car_maker_task/src/main/java/com/epam/cars/H2/H2CarRepository.java

```
@@ -92,35 +98,23 @@ private Car toCar(final ResultSet rs) throws SQLException {
 
     @Override
     public void saveCar(Car car) {
-        try (Connection con = DriverManager.getConnection(connect.getUrl(),
-                connect.getUser(), connect.getPassword())) {
-            pstmt = con.prepareStatement("Insert into CAR Values(?,?,?,?)");
-            pstmt.setString(1, car.getModel());
-            pstmt.setString(2, car.getColor());
-            pstmt.setInt(3, car.getYear());
-            pstmt.setLong(4, car.getMaker().getId());
-            pstmt.execute();
-        } catch (SQLException ex) {
+        try {
+            jdbcTemplate.update("Insert into CAR Values(?,?,?,?,?)",
+                    ++lastCarId, car.getModel(), car.getColor(), car.getYear(), car.getMaker().getId());
+        } catch (DataAccessException ex) {
             LOG.error("error connection or query ", ex);
```

This is not good that you silently fail to save user and only log the error... but user can't know if save is success or failure...

It makes sense to propagate exception up to caller - so he can handle the error.